frontend/src/metabase/entities/bookmarks.js

```diff
@@ -17,6 +21,34 @@ const Bookmarks = createEntity({
       return BookmarkApi[type].delete({ id });
     },
   },
+  objectSelectors: {
+    getIcon,
+  },
 });
 
+function defineIconGetter(type) {
+  const getters = {
+    card: Question,
+    collection: Collection,
+    dashboard: Dashboard,
+  };
+
+  return getters[type];
+}
+
+export function getIcon(bookmark) {
+  const { type } = bookmark;
+
+  const getter = defineIconGetter(type);
+
+  const { color: iconColor, name, tooltip } = getter.objectSelectors.getIcon(
+    bookmark,
+  );
+
+  const treatedColor = type === "card" ? color("brand") : iconColor;
+  const opacity = tooltip ? 1 : 0.5;
+
+  return { name, color: treatedColor, opacity };
+}
```

Comment:
Please, let's manage as much styling as possible inside components instead of entities. Entities `getIcon` functions were extended for business-logic reasons and not styling.

Reply:
👍, updated